### ISSUE TITLE
ZAP should provide ability to pause autoscan and continue scan after pause

### ISSUE THREAD

**Issue by JuraLys** [zaproxy/zaproxy#8271]
### Is your feature request related to a problem? Please describe.

ZAP should provide ability to pause autoscan and continue scan after pause

### Describe the solution you'd like

ZAP should provide ability to pause autoscan and continue scan after pause

### Describe alternatives you've considered

no

### Would you like to help fix this issue?

- [X] Yes

**Comment by thc202** [comment]
You can pause the spiders or active Scan from the respective tabs.

**Comment by JuraLys** [comment]
So it will be good to duplicate Pause button near Stop button under URL field.

**Comment by psiinon** [comment]
Yeah, have added the `good second issue` label. Not something that I would address as a priority, but happy for someone else to imlement it.
Note that the AJAX Spider cannot currently be paused, and I'd consider that as out of scope for this issue.

**Comment by vishalvivekm** [comment]
Hi @psiinon,
Starting my contribution journey with ZAP, can you please assign this issue to me?

**Comment by psiinon** [comment]
@vishalvivekm done 😁

**Comment by amitpanwar789** [comment]
@vishalvivekm are you working on it ? if not then i would like to work on it .

**Comment by jessemchung** [comment]
Any updates on this?  Am curious if me and a friend could try to make it work

**Comment by kingthorin** [comment]
Go for it.

**Comment by psiinon** [comment]
It should be ralatively easy - just look at the code that stops the scans and duplicate that for pausing.
Note that the AJAX Spider doesnt currently support pausing, so you should disable the pause button while that is running.

**Comment by jessemchung** [comment]
Okay I will try to figure this out.

**Comment by jessemchung** [comment]
@psiinon I'm sorry, I'm very new to this and my friend who I planned to work on this with has more or less dropped out at the moment.  I am uncertain how best to start, there used to be a quickstart?  I cannot access it and I am having trouble getting an active scan started.  Sorry for the trouble

**Comment by jessemchung** [comment]
@psiinon 
Yeah, I tried setting it up again, but the green plus mark still will not show anything.  I just see "break" as an option
<img width="1790" alt="Screenshot 2024-09-12 at 12 59 00 AM" src="https://github.com/user-attachments/assets/dcd81db5-7882-4793-acb5-00dc18b9ad71">

I tried reinstalling the quickstart extension like so which seems to have worked according to the IDE.  But still nothing is appearing.  

<img width="1790" alt="Screenshot 20
<img width="1305" alt="Screenshot 2024-09-12 at 1 00 48 AM" src="https://github.com/user-attachments/assets/94e0a36b-0b26-4d2a-a289-e6ca17006964">
24-09-12 at 12 59 00 AM" src="https://github.com/user-attachments/assets/15e290cd-e717-4fab-94f7-7065fcda18ad">

Do we need to clone the following core help and hud?  I wasn't sure if that was important or not but it seems irrelevent
git clone https://github.com/zaproxy/zap-hud.git
git clone https://github.com/zaproxy/zap-core-help.git

Sorry for the trouble

**Comment by thc202** [comment]
Better move this to the dev mailing list since this is not directly about the issue: https://groups.google.com/g/zaproxy-develop/

For this specific case the HUD and help is not necessary. You should tell how you are installing the Quick Start add-on.

**Comment by jessemchung** [comment]
@kingthorin 
We (@In-Verse ) got it working (some hiccups still) but we just want to confirm.  The only thing that we should be fixing is this page that is a part of the quickstart right?  Meaning most of our fixes will be in the zap-extensions repo.

<img width="1792" alt="Screenshot 2024-09-13 at 5 45 42 PM" src="https://github.com/user-attachments/assets/8fb77770-47e2-427b-8474-0fd1c2fdd061">

**Comment by thc202** [comment]
That's right.

**Comment by jessemchung** [comment]
Hello,

so we've taken time to investigate this issue and this is what we sort of think needs to be done.  We could probably get a wireframe up at some point but for now, this is the idea in text.

- Add pause button
- This button, when pressed should pause the scan like it does the button below does with the active scans
- New text shows up (for example, when stop is pressed it says "Attack stopped manually", we want it to say "Attack paused manually")
- Of note, pressing the pause and stop button on the bottom has no effect on the quickstart window dialogue
- Clicking the pause button could either

1. change the pause button to resume (maybe also change the name of the start button to restart?)
2. or disable the pause button and allow clicking the start button to resume (we would need to possibly set a hard width so that the button doesn't change sizes between the two)

- We are currently investigating the brute force extension as well as the lower pause button to see what we should imitate

**Comment by thc202** [comment]
> Add pause button

IMO the Attack button could be changed to Pause/Resume once started (if the current scan allows it).

> New text shows up

I'd say "<scan> Paused".

> Of note, pressing the pause and stop button on the bottom has no effect on the quickstart window dialogue

Currently if stopped it has the effect of continuing the process, either going to next scan or showing the results. IMO we should change the state to paused in the Automated Scan if the user pauses one of the scans in their own panels.

> We are currently investigating the brute force extension as well as the lower pause button to see what we should imitate

I'd suggest the standard spider (though it's a core class that you need to check).

**Comment by alhd12** [comment]
is this issue already fixed?

**Comment by thc202** [comment]
No.
